src/irr/asset/CGLILoader.cpp

```diff
@@ -0,0 +1,457 @@
+/*
+MIT License
+Copyright (c) 2019 AnastaZIuk
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#include "CGLILoader.h"
+
+#ifdef _IRR_COMPILE_WITH_GLI_LOADER_
+
+#ifdef _IRR_COMPILE_WITH_GLI_
+#include "gli/gli.hpp"
+#else
+#error "It requires GLI library"
+#endif
+
+namespace irr
+{
+	namespace asset
+	{
+		inline std::pair<E_FORMAT, ICPUImageView::SComponentMapping> getTranslatedGLIFormat(const gli::texture& texture, const gli::gl& glVersion);
+		inline void assignGLIDataToRegion(void* regionData, const gli::texture& texture, const uint16_t layer, const uint16_t face, const uint16_t level, const uint64_t sizeOfData);
+
+		asset::SAssetBundle CGLILoader::loadAsset(io::IReadFile* _file, const asset::IAssetLoader::SAssetLoadParams& _params, asset::IAssetLoader::IAssetLoaderOverride* _override, uint32_t _hierarchyLevel)
+		{
+			if (!_file)
+				return {};
+
+			const char* filename = _file->getFileName().c_str();
+			gli::texture texture = gli::load(filename);
```

Comment:
this won't work, you need to be able to read from virtual files... etc. so find a GLI function or provide one that overrides `fwrite` or whatever its using internally.

Maybe there's already something that uses an `ostream` or `istream` ?

Reply:
why? I think it will https://github.com/g-truc/ovtsa/blob/master/external/gli/core/load.inl#L12

Reply:
because you're supposed to read from the `IReadFile` that is passed as argument, you do realize that it doesn't have to be a file in the filesystem, and could be a virtual file in an archive, OR just a pile of data in RAM that mimicks a file?

Reply:
well I didn't know that, so I will adjust it